File: python/leetcode/longest-continuous-increasing-subsequence.py
class Solution:
    def findLengthOfLCIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        longest = 0
        if len(nums) == 0:
            return longest
        last = nums[0]
        longest += 1
        this_len = longest
        for n in nums:
            if n > last:
                this_len += 1
            else:
                if this_len > longest:
                    longest = this_len
                this_len = 1
            last = n
        if this_len > longest:
            longest = this_len
        return longest

File: java/leetcode/longest-continuous-increasing-subsequence.java
class Solution {
    public int findLengthOfLCIS(int[] nums) {
        int longest = 0;
        if (nums.length == 0) { return longest; }
        int last = nums[0];
        longest++;
        int thisLen = longest;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > last) {
                thisLen++;
            } else {
                if (thisLen > longest) { longest = thisLen; }
                thisLen = 1;
            }
            last = nums[i];
        }
        if (thisLen > longest) { longest = thisLen; }
        return longest;
    }
}
